lib/zip/entry_set.rb

```diff
@@ -13,8 +13,10 @@ def include?(entry)
       @entry_set.include?(to_key(entry))
     end
 
-    def find_entry(entry)
-      @entry_set[to_key(entry)]
+    def find_entry(entry, case_sensitively = true)
+      return @entry_set[to_key(entry)] if case_sensitively
+      entry = @entry_set.find { |k, _| k.downcase == to_key(entry).downcase }
```

Comment:
@rastarobot, you are 100% correct that "hash keys need to be downcased too", but aren't the hash entry keys formed by calling `to_key` already?

Something like 
```
def add_file_to_hash(name, entry)
  @hash[to_key(name)] = entry
   ...
end
```
That's the entire point why I'm telling you to modify `to_key`, not `find`.